integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -812,25 +759,18 @@ public int write(String path, Pointer buf, @size_t long size, @off_t long offset
    * Convenience internal method to remove files or directories.
    *
    * @param path The path to remove
-   * @param mustBeFile When true, returns an error when trying to
-   *                   remove a directory
+   * @param isDirectory true when removing a directory, false otherwise
    * @return 0 on success, a negative value on error
    */
-  private int rmInternal(String path, boolean mustBeFile) {
+  private int rmInternal(String path, boolean isDirectory) {
     final AlluxioURI turi = mPathResolverCache.getUnchecked(path);
 
     try {
-      if (!mFileSystem.exists(turi)) {
-        LOG.error("File {} does not exist", turi);
-        return -ErrorCodes.ENOENT();
+      DeleteOptions options = DeleteOptions.defaults();
+      if (isDirectory) {
+        options.setRecursive(true);
       }
-      final URIStatus status = mFileSystem.getStatus(turi);
-      if (mustBeFile && status.isFolder()) {
-        LOG.error("File {} is a directory", turi);
-        return -ErrorCodes.EISDIR();
-      }
-
-      mFileSystem.delete(turi);
+      mFileSystem.delete(turi, options);
```

Comment:
I don't think we ever want to set the recursive flag? Unfortunately, if `isDirectory` is true, I think we need to do 1 extra RPC, or modify our options to be finer grained.

Delete will delete files and empty directories. Delete with recursive will in addition delete non-empty dirs and their children.

I think what FUSE wants is to delete only files, and another command to delete only empty directories?

Reply:
In `libfuse`, the explaination of `unlink` and `rmdir` is
```
	/** Remove a file */
	int (*unlink) (const char *);

	/** Remove a directory */
	int (*rmdir) (const char *);
```
It does not say for sure that rmdir is removing an empty directory.
Different filesystems may have different behaviors. Some may call `unlink` to delete files and rmdir to remove empty folder, but some may directly use `rmdir` to remove directory.

Reply:
As implemented, our semantic is
`unlink` deletes files only.
`rmdir` deletes directories (empty and non-empty).

I am unsure about `rmdir`, on most OS this will not delete non-empty directories. Are you saying the FUSE interface is different, how do other implementations of FUSE filesystems interpret `rmdir`?

Reply:
In the end of tests like filebench, filesystem may want to directly delete all the test data and trigger `rmdir` to delete non-empty directories. 

Libfuse API documentation is actually quite complete and detailed. If `rmdir` only delete non-empty directory, they will definitely say in the method comment.

I will only set delete options in `rmdir` to simplify the logic of `unlink`

Reply:
Will open a linux instance and test with remove non-empty directories. If it can pass my manually test and autobots test. I am fine with directly using delete without options. If users find new cases where `rmdir` will delete the non-empty directory. We can simplify add it back

Reply:
I may not need to test it. Our old form is directly delete folders/files and no errors found. Will change to not add recursive. Thanks